core/include/msg_bus.h

```diff
@@ -68,19 +74,48 @@ void msg_bus_init(msg_bus_t *bus);
  * The `type` field of the`msg_t` also encodes the message bus ID,
  * so use this function to get the real 5 bit message type.
  *
+ * If the message was not send over a bus, this will return the
+ * original message ID.
+ *
  * @param[in] msg           A message that was received over a bus
  *
  * @return                  The message type
  */
-static inline uint8_t msg_bus_get_type(const msg_t *msg)
+static inline uint16_t msg_bus_get_type(const msg_t *msg)
+{
+    if (msg->sender_pid & MSB_BUS_PID_FLAG) {
+        return msg->type & 0x1F;
+    } else {
+        return msg->type;
+    }
+}
+
+/**
+ * @brief Get the sender PID of a message bus message.
+ *
+ * The `sender_pid` field of the`msg_t` has a flag bit set
+ * to indicate the message was sent over a bus, thus it should
+ * not be used directly.
+ *
+ * @param[in] msg           A message that was received over a bus
+ * @param[out] from_bus     Optional parameter that is `true` if the message
+ *                          was received over a bus.
+ *
+ * @return                  The sender pid
+ */
+static inline kernel_pid_t msg_bus_get_sender_pid(const msg_t *msg, bool *from_bus)
```

Comment:
I would personally drop the `from_bus` argument and provide a second helper function to check if the message is a "bus message" or a "regular message". That might be two lines of code instead of one in some cases, but two self-explaining lines of code are IMO better than one magic argument I would have to look up in the doc every time.

Reply:
tbh I can't think of a use case for such a function that is not also covered by `msg_is_from_bus()` - I'll just drop it.